pyam/core.py

```diff
@@ -686,7 +686,10 @@ def timeseries(self, iamc_index=False):
         if self.empty:
             raise ValueError('this `IamDataFrame` is empty')
 
-        df = self._data.unstack(level=self.time_col).rename_axis(None, axis=1)
+        s = self._data
+        if iamc_index:
```

Comment:
Note that an IamDataFrame could also use `time` instead of `year` or a custom index like `['model', 'scenario', 'version']`- in this case, just dropping the extra-columns will not yield the expected IAMC-index.

Reply:
I have not used sub-yearly dataframes yet. Could you point me at an example? And how is the iamc_index defined in that case?

Reply:
Unfortunately, we don't have any decent documentation yet for these formats...

1. continuous-time format: use a column 'time' (instead of 'year') with format `datetime`, e.g.

    | **model** | **scenario** | **region** | **variable**   | **unit** | **time** | **value** |
    |-----------|--------------|------------|----------------|----------|----------|----------|
    | MESSAGE   | CD-LINKS 400 | World      | Primary Energy | EJ/y     |    2020-03-09 13:10 |    500.7 |
    | ...       | ...          | ...        | ...            | ...      |      ... |      ... |      ... |

    Many features are tested for this format, see https://github.com/IAMconsortium/pyam/blob/7bcedfaf5cec135ab0f9cb58c310cb694ddf6c7f/tests/conftest.py#L129

    This should raise a `ValueError("Cannot use iamc-index with continuous-time data format!")`.

2. subannual-column: for a time representation that can't be handled as continuous-time (e.g., representative time-steps like `summer-day`), we use a 'subannual' column.

    | **model** | **scenario** | **region** | **variable**   | **unit** | **subannual** | **2005** | **2010** | **2015** |
    |-----------|--------------|------------|----------------|----------|---------------|----------|----------|----------|
    | MESSAGE   | CD-LINKS 400 | World      | Primary Energy | EJ/y     |  winter-day |  462.5 |    500.7 |      ... |
    | ...       | ...          | ...        | ...            | ...      |      ... |      ... |      ... |

   In this case (ie., df.time_col == "year"), `subannual` will just be dropped as part of `drop level(self.extra_cols
)`

Reply:
Thanks!

Regarding 2:
> In this case (ie., df.time_col == "year"), subannual will just be dropped as part of drop level(self.extra_cols )

I guess that is also the desired behaviour, is it? It's quite likely to run into the "Data with IAMC-index has duplicates"  `ValueError`.

Reply:
Yes, this was/is intentional... The rationale at the time: we may introduce fancy methods that add extra-columns, like open-scm adding a column "climate_model" to track which model was used for the temperature assessment. Then, it would be helpful for users to have an easy method to get rid of the extra-columns, for example to pass the timeseries data another script or tool that doesn't have the flexibility of pyam and can only handle a "strict" IAMC format.

If that works by simple dropping the extra-col (using `iamc_index=True`, fine - otherwise, the user has to come up with something smarter...

I'd keep that option for now (even if only mildly useful), and maybe we add a `drop()´ at a later point to replace it...

Reply:
Fine!